Monika After Story/game/script-farewells.rpy

```diff
@@ -546,7 +546,41 @@ label bye_prompt_sleep:
 
     if 20 <= curr_hour < 24:
         # decent time to sleep
-        if mas_isMoniNormal(higher=True):
+        if mas_isMoniEnamored(higher=True) and persistent._mas_first_kiss is not None and random.randint(1, 3) == 1:
```

Comment:
This check should be replaced with `mas_shouldKiss`, it handles random chance, the first kiss check, and the affection check all in one